src/pycram/worlds/multiverse.py

```diff
@@ -0,0 +1,678 @@
+        return [joint for joint in joints if self.joint_has_actuator(joint)]
+
+    def _set_multiple_joint_positions_without_controller(self, joint_positions: Dict[Joint, float]) -> None:
+        """
+        Set the positions of multiple joints in the simulator without using the controller.
+
+        :param joint_positions: The dictionary of joints and positions.
+        """
+        joints_data = {joint.name: {self.get_joint_position_name(joint): [position]}
+                       for joint, position in joint_positions.items()}
+        self.writer.send_multiple_body_data_to_server(joints_data)
+
+    def _set_multiple_joint_positions_using_controller(self, joint_positions: Dict[Joint, float]) -> bool:
+        """
+        Set the positions of multiple joints in the simulator using the controller.
+
+        :param joint_positions: The dictionary of joints and positions.
+        """
+        controlled_joints_data = {self.get_actuator_for_joint(joint):
+                                      {self.get_joint_cmd_name(joint.type): [position]}
+                                  for joint, position in joint_positions.items()}
+        self.joint_controller.send_multiple_body_data_to_server(controlled_joints_data)
+        return True
+
+    def get_joint_position(self, joint: Joint) -> Optional[float]:
+        joint_position_name = self.get_joint_position_name(joint)
+        data = self.reader.get_body_data(joint.name, [joint_position_name])
+        if data is not None:
+            return data[joint_position_name.value][0]
+
+    def get_multiple_joint_positions(self, joints: List[Joint]) -> Optional[Dict[str, float]]:
+        joint_names = [joint.name for joint in joints]
+        data = self.reader.get_multiple_body_data(joint_names, {joint.name: [self.get_joint_position_name(joint)]
+                                                                for joint in joints})
+        if data is not None:
+            return {name: list(value.values())[0][0] for name, value in data.items()}
+
+    @staticmethod
+    def get_joint_cmd_name(joint_type: JointType) -> MultiverseJointCMD:
+        """
+        Get the attribute name of the joint command in the Multiverse from the pycram joint type.
+
+        :param joint_type: The pycram joint type.
+        """
+        return MultiverseJointCMD.from_pycram_joint_type(joint_type)
+
+    def get_link_pose(self, link: Link) -> Optional[Pose]:
+        return self._get_body_pose(link.name)
+
+    def get_multiple_link_poses(self, links: List[Link]) -> Dict[str, Pose]:
+        return self._get_multiple_body_poses([link.name for link in links])
+
+    def get_object_pose(self, obj: Object) -> Pose:
+        if obj.obj_type == ObjectType.ENVIRONMENT:
+            return Pose()
+        return self._get_body_pose(obj.name)
+
+    def get_multiple_object_poses(self, objects: List[Object]) -> Dict[str, Pose]:
+        env_objects = [obj for obj in objects if obj.obj_type == ObjectType.ENVIRONMENT]
```

Comment:
because they are always at 0, 0, 0, they are loaded from the .muv file, they are not loaded during runtime , also environment objects tend to be very big and has lots of links which may cause issues with multiverse that's why setting environment poses is not supported as well.

Reply:
Done